pkg/scanner/scanner.go

```diff
@@ -195,16 +264,276 @@ func (s *Scanner) processNextWorkItem(ctx context.Context) bool {
+	}
+	patchBytes, err := json.Marshal(patchData)
+	if err != nil {
+		return fmt.Errorf("failed to marshal annotation patch data for PV %s: %w", pvName, err)
+	}
+	klog.V(6).Infof("Patching PV %s annotations with: %s", pvName, string(patchBytes))
+	_, err = s.kubeClient.CoreV1().PersistentVolumes().Patch(ctx, pvName, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Warningf("Failed to patch PV %s because it was not found.", pvName)
+			return nil
+		}
+		return fmt.Errorf("failed to patch PV %s annotations: %w", pvName, err)
+	}
+	klog.V(6).Infof("Successfully patched annotations for PV %s", pvName)
+	return nil
+}
+
+// updatePVScanResult updates the PV annotations with the results of a bucket scan.
+// It sets the status, number of objects, total size, HNS status, and last updated time.
+func (s *Scanner) updatePVScanResult(ctx context.Context, pv *v1.PersistentVolume, info *bucketInfo, status string) error {
+	annotationsToUpdate := map[string]*string{
+		annotationStatus:          stringPtr(status),
+		annotationNumObjects:      int64Ptr(info.numObjects),
+		annotationTotalSize:       int64Ptr(info.totalSizeBytes),
+		annotationLastUpdatedTime: stringPtr(timeNow().UTC().Format(time.RFC3339)),
+		annotationHNSEnabled:      boolPtr(info.isHNSEnabled),
+	}
+	klog.Infof("Updating PV %s with scan result: %+v, status: %s", pv.Name, info, status)
+	err := s.patchPVAnnotations(ctx, pv.Name, annotationsToUpdate)
+	if err != nil {
+		klog.Errorf("Failed to update annotations on PV %s with status %s: %v", pv.Name, status, err)
+		return err
+	}
+	klog.Infof("Successfully updated annotations on PV %s with status %s", pv.Name, status)
 	return nil
 }
 
+// checkPVRelevance determines if a PersistentVolume is relevant for scanning.
+// A PV is relevant if it uses the gcsfuse CSI driver and its StorageClass
+// has a workloadType parameter set to inference, training, or checkpointing.
+// The PV is relevant if the current time - last scan time > scan TTL
+// This function returns a bucketInfo with the bucket name and the directory if
+// relevant, otherwise, it will return nil and any error.
+func (s *Scanner) checkPVRelevance(pv *v1.PersistentVolume) (*bucketInfo, error) {
+	if pv == nil {
+		return nil, nil
+	}
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csiDriverName || pv.Spec.CSI.VolumeHandle == "" {
+		return nil, nil
+	}
+
+	scName := pv.Spec.StorageClassName
+	if scName == "" {
+		return nil, nil
+	}
+	sc, err := s.scLister.Get(scName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get StorageClass %s: %w", scName, err)
+		}
+		klog.Warningf("StorageClass %s not found for PV %s", scName, pv.Name)
+		return nil, nil
+	}
+
+	scParams := sc.Parameters
+	if scParams == nil {
+		return nil, nil
+	}
+	workloadType, ok := scParams[paramWorkloadTypeKey]
+	if !ok {
+		return nil, nil
```

Comment:
maybe put a log with a higher verbosity value, such that if we need to debug, we can enable them